File: Estudos/M1P1-25.2/Ex4.r
set.seed(123)

custo <- function(area){
    return(volume(area)*250)
}

volume <- function(area){
    return(area*6)
}

gera_coord <- function(){
    x = runif(1,0,1)
    y = runif(1,0,1)

    return(c(x,y))
}

estima_area <- function(){
    coord <- gera_coord()
    x <- coord[1]
    y <- coord[2]
    if(x>=0.5 && y >= 0.5 && (x**2 + y**2 >= 1)) {
        return(1)
    }
    return(0)
}

dentro = 0
for(i in 1:10000){
    dentro <- dentro + estima_area()
}

area = dentro / 10000
cat("Area estimada:", area)
cat("\nCusto estimado: ", custo(area))

File: Estudos/M1P1-25.2/Ex4.py
import random

random.seed(123)

def custo(area_base_cinza):
    return volume(area_base_cinza)*250

def volume(area):
    return area*6

def gera_coord():
    x = random.uniform(0,1)
    y = random.uniform(0,1)

    return(x,y)

def estima_area():
    x,y = gera_coord()
    if x >= 0.5 and y >= 0.5 and (x**2 + y**2 >= 1):
        return 1
    return 0

dentro= 0
for _ in range(10000):
    dentro += estima_area()

area = dentro / 10000
print("Área estimada:", area)
print("Custo estimado:", custo(area))